### PR TITLE
Make all cheatcodes use one shared `internal_calls`. Fix `send_message_to_l2` cheatcode.

### LINKED ISSUE #1065: Fix `send_message_to_l2` cheatcode — `list assignment index out of range`
A similar issue occurred for the `deploy` cheatcode https://github.com/software-mansion/protostar/issues/398.

https://discord.com/channels/793094838509764618/964116575387021413/1033112627590737951

### PR DESCRIPTION
**Pull request by kasperski95**
Fix crash caused by `send_message_to_l2` cheatcode. Fix the problem in the `Cheatcode` class to avoid reintroducing this problem by new cheatcodes.

Fixes #1065

### CODE REVIEW

**Comment by Arcticae** [comment]
This seems a bit weird since we were using this `internal_calls` thing for some reason - why are we deleting it now? Something in the cairo-lang changed in this matter?

**Comment by mkaput** [comment]
I don't understand from the PR description: what was the issue? Could you please clarify? I don't know what to focus on :(

**Comment by kasperski95** [comment]
> This seems a bit weird since we were using this internal_calls thing for some reason - why are we deleting it now? Something in the cairo-lang changed in this matter?

> I don't understand from the PR description: what was the issue? Could you please clarify? I don't know what to focus on :(

We are not deleting it. Instead, `internal_calls` are passed with `Cheatcode.SysCallDependencies`.

Multiple syscall handlers break StarkNet internals. Originally, the issue was caused by the deploy cheatcode. However, this problem appeared once again with the introduction of the `send_message_to_l2`. This PR fixes this problem in a more generic way.

**Comment by mkaput** [comment]
Please rewrite PR title and description to focus on shifting `internal_calls` passing to different place then 😃

File: tests/integration/cheatcodes/send_message_to_l2/external_contract_with_l1_handler.cairo [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
 from starkware.cairo.common.cairo_builtins import HashBuiltin
 
-
 @storage_var
 func state() -> (res: felt) {
 }
 
+@event
+func fake_event() {
+}
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/cheatcodes/send_message_to_l2/external_contract_with_l1_handler.cairo:10]: It was needed to replicate the reported problem.

**Review by Arcticae — APPROVED** [review]

File: tests/integration/cheatcodes/send_message_to_l2/external_contract_with_l1_handler.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkaput) [reply tests/integration/cheatcodes/send_message_to_l2/external_contract_with_l1_handler.cairo:10]: Could you please add a comment explaining why is it important next to it?

[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/cheatcodes/send_message_to_l2/external_contract_with_l1_handler.cairo:10]: Done

**Pull request merged**